File: src/ros2_bridge/mappers/control_msgs/motion_argument.rs
//! Typed mapper for `control_msgs/msg/MotionArgument`.

use crate::ros2_bridge::mapper::TypedTopicMapper;

pub(crate) fn motion_argument_to_bus(msg: ros_env::control_msgs::msg::MotionArgument) -> crate::control_msgs::msg::v1::MotionArgument {
    crate::control_msgs::msg::v1::MotionArgument {
        name: crate::ros2_bridge::mappers::convert::from_ros_string(msg.name),
        value: msg.value,
    }
}

pub(crate) fn motion_argument_to_ros(bus: crate::control_msgs::msg::v1::MotionArgument) -> ros_env::control_msgs::msg::MotionArgument {
    ros_env::control_msgs::msg::MotionArgument {
        name: crate::ros2_bridge::mappers::convert::to_ros_string(bus.name),
        value: bus.value,
    }
}

#[derive(Clone, Copy, Debug, Default)]
pub struct ControlMsgsMotionArgumentMapper;

impl TypedTopicMapper for ControlMsgsMotionArgumentMapper {
    type Ros = ros_env::control_msgs::msg::MotionArgument;
    type Bus = crate::control_msgs::msg::v1::MotionArgument;

    fn type_name(&self) -> &'static str {
        "control_msgs/msg/MotionArgument"
    }

    fn ros_to_bus(&self, msg: Self::Ros) -> crate::errors::Result<Self::Bus> {
        Ok(motion_argument_to_bus(msg))
    }

    fn bus_to_ros(&self, msg: Self::Bus) -> crate::errors::Result<Self::Ros> {
        Ok(motion_argument_to_ros(msg))
    }
}

File: bindings/python/robot_bus/ros2_bridge/mappers/control_msgs/motion_argument.py
"""Generated mapper for `control_msgs/msg/MotionArgument`."""

from __future__ import annotations

from robot_bus.ros2_bridge.mappers import _convert

_BusMsg = None


def _bus_cls():
    global _BusMsg
    if _BusMsg is None:
        from robot_bus.control_msgs.msg.v1 import MotionArgument as BusMsg

        _BusMsg = BusMsg
    return _BusMsg


def motion_argument_to_bus(msg):
    BusMsg = _bus_cls()
    bus = BusMsg()
    bus.name = str(msg.name)
    bus.value = msg.value
    return bus


def motion_argument_to_ros(bus):
    from control_msgs.msg import MotionArgument as RosMsg

    out = RosMsg()
    out.name = str(bus.name)
    out.value = bus.value
    return out


class ControlMsgsMotionArgumentMapper:
    _ros_type = None

    def ros_msg_type(self):
        cls = type(self)
        if cls._ros_type is None:
            from control_msgs.msg import MotionArgument as RosMsg

            cls._ros_type = RosMsg
        return cls._ros_type

    def ros_to_bus(self, msg) -> bytes:
        return motion_argument_to_bus(msg).SerializeToString()

    def bus_to_ros(self, payload: bytes):
        BusMsg = _bus_cls()
        bus = BusMsg()
        bus.ParseFromString(payload)
        return motion_argument_to_ros(bus)
